packages/itwinui-react/src/core/utils/hooks/useMediaQuery.ts

```diff
@@ -3,36 +3,23 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
-import { getWindow } from '../functions/index.js';
-import { useLayoutEffect } from './useIsomorphicLayoutEffect.js';
+import { useSyncExternalStore } from './useSyncExternalStore.js';
 
 export const useMediaQuery = (queryString: string) => {
-  const [matches, setMatches] = React.useState<boolean>();
-
-  useLayoutEffect(() => {
-    const mediaQueryList = getWindow()?.matchMedia?.(queryString);
-    const handleChange = ({ matches }: MediaQueryListEvent) =>
-      setMatches(matches);
-
-    if (mediaQueryList != undefined) {
-      setMatches(mediaQueryList.matches);
-      try {
-        mediaQueryList.addEventListener('change', handleChange);
-      } catch {
-        // Safari 13 fallback
-        mediaQueryList.addListener?.(handleChange);
-      }
-    }
-
-    return () => {
-      try {
-        mediaQueryList?.removeEventListener('change', handleChange);
-      } catch {
-        // Safari 13 fallback
-        mediaQueryList?.removeListener?.(handleChange);
-      }
-    };
+  const getSnapshot = React.useCallback(() => {
+    return typeof window !== 'undefined'
+      ? window.matchMedia?.(queryString).matches
+      : undefined;
   }, [queryString]);
 
-  return !!matches;
+  const subscribe = React.useCallback(
+    (onChange: () => void) => {
+      const mediaQueryList = window.matchMedia?.(queryString);
+      mediaQueryList?.addEventListener?.('change', onChange);
```

Comment:
The Safari fallback is no longer needed? If so, should it be mentioned in a changeset?

Reply:
nah, it's been supported since safari 14, which is long enough to not be important.

and it's still using `?.` so it will still work correctly for first mount, it just will never update.